File: MG.WebHost/Models/Page.cs
using System.Collections.Generic;

namespace MG.WebHost.Models
{
    public record Page<T> where T : class
    {
        public bool NextPageAvailable => PageNumber * PageSize < Count;
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public int Count { get; set; }
        public IEnumerable<T> Elements { get; set; }
    }
}

File: WebApp/src/app/models/page.ts
export class Page<T> implements PageOptions {
  nextPageAvailable: boolean = false;
  pageNumber: number = 0;
  pageSize: number = 0;
  count: number = 0;
  elements: T[] = []
}

export interface PageOptions {
  pageNumber: number;
  pageSize: number;
  count: number;
}
